MovieApp/DataLayer/DataSources/MovieLocalDataSource/MovieLocalDataSource.swift

```diff
@@ -0,0 +1,70 @@
+import RealmSwift
+import Combine
+
+class MovieLocalDataSource: MovieLocalDataSourceProtocol {
+
+    var popularMovies: AnyPublisher<[MovieDataSourceModel], Error> {
+        movies
+            .map { $0.filter { $0.categories.contains(.popular) } }
```

Comment:
for the next pr, you should investigate if this can be filtered via Realm filter.